### PR TITLE
Add support for Azure Scale Sets

### PR DESCRIPTION
**Pull request by cberner**
This will replace the current Azure integration, which relies on a custom backend

### CODE REVIEW

File: autoscaler/azure.py [hunk 1]
```diff
@@ -122,6 +180,118 @@ def _get_azure_class(type_):
...
+        self.resource_group = resource_group
+
+        self.selectors = dict(self.tags)
+        # HACK: for matching node selectors
+        self.selectors['azure/type'] = self.instance_type
+        self.selectors['azure/region'] = self.region
+        self.selectors['azure/class'] = _get_azure_class(self.instance_type)
+
+        self.min_size = 0
+        self.max_size = 1000
```
Comment (vicki-c) [inline autoscaler/azure.py]: would this hard-coded value mean that we won't need more than 10 scalesets?

File: autoscaler/kube.py [hunk 2]
```diff
@@ -88,6 +88,13 @@ def __repr__(self):
         return str(self)
 
 
+def reverse_bytes(value):
```
Comment (vicki-c) [inline autoscaler/kube.py]: 😅

File: autoscaler/azure.py (hunk 1 above)
Comment (cberner) [inline autoscaler/azure.py]: Yes, well, actually I just discovered that Terraform doesn't support Managed Disks for Scale Sets, so we can only have 40 VMs per Scale Set (I'll change the constant) :( So we'd need 25. I just copied this over from the old Azure integration, which had a limit of 1000

**Comment by cberner** [comment]
Added one more commit to batch delete instances

**Pull request merged**